modules/pool/lib/pool-on-chain-data.service.ts

```diff
@@ -90,37 +41,12 @@ export class PoolOnChainDataService {
             },
         });
 
-        const multicall = new Multicaller3(ComposableStablePoolAbi);
-
-        filteredPools.forEach((pool) => {
-            if (!(pool.type === 'ELEMENT')) {
-                multicall.call(`${pool.id}.pausedState`, pool.address, 'getPausedState');
-            }
-            if (
-                pool.type !== 'LIQUIDITY_BOOTSTRAPPING' && // exclude all LBP
-                pool.type !== 'META_STABLE' && // exclude meta stable
-                pool.type !== 'ELEMENT' // exclude element
-            ) {
-                multicall.call(`${pool.id}.inRecoveryMode`, pool.address, 'inRecoveryMode');
-            }
-        });
-
-        const multicallResult = (await multicall.execute()) as Record<string, MulticallPoolStateExecuteResult>;
-
-        const poolStatusResults: PoolStatusResult = {};
+        const state = await fetchOnChainPoolState(filteredPools, 1024);
 
         const operations = [];
         for (const pool of filteredPools) {
+            const { isPaused, isInRecoveryMode } = state[pool.id];
             if (pool.dynamicData) {
-                let isPaused = false;
-                let isInRecoveryMode = false;
-
-                if (multicallResult[pool.id] && multicallResult[pool.id].inRecoveryMode) {
-                    isInRecoveryMode = multicallResult[pool.id].inRecoveryMode;
-                }
-                if (multicallResult[pool.id] && multicallResult[pool.id].pausedState) {
-                    isPaused = multicallResult[pool.id].pausedState.paused;
-                }
                 operations.push(
                     prisma.prismaPoolDynamicData.update({
                         where: { id_chain: { id: pool.id, chain: networkContext.chain } },
```

Comment:
any chance we could also get rid of the network context while we're refactoring that?

Reply:
good point, moving it up to the pool service